packages/types/src/extrinsic/SignerPayload.spec.ts

```diff
@@ -130,11 +134,11 @@ describe('SignerPayload', (): void => {
     expect(payload.blockHash.toHex()).toEqual(TEST.blockHash);
     expect(payload.nonce.eq(TEST.nonce)).toBe(true);
     expect(payload.tip.eq(TEST.tip)).toBe(true);
-    expect(payload.assetId.isSome && payload.assetId
-      ?.eq(registry.createType('MultiLocation', {
+    expect(u8aToHex(payload.assetId?.toU8a()))
```

Comment:
We could just use `.toHex()` and make it simpler:
```
expect(payload.assetId?.toHex())
      .toEqual(registry.createType('Option<MultiLocation>', {
        // eslint-disable-next-line sort-keys
        parents: 0, interior: { X2: [{ palletInstance: 50 }, { generalIndex: 123 }] }
      }).toHex());
      ```

Reply:
This doesn't work -  `toHex` will unwrap the value if possible... therefore we have to do the above workaround.

Reply:
You are right, it's in the definition `toHex()` in `Option`:
```
public toHex (): HexString {
    // This attempts to align with the JSON encoding - actually in this case
    // the isSome value is correct, however the `isNone` may be problematic
    return this.isNone
      ? '0x'
      : u8aToHex(this.toU8a().subarray(1));
  }
  ```